tests/transports/test_all_plugins.py

```diff
@@ -1358,10 +1362,135 @@ def test_exec_with_stdin_filelike(self, custom_transport):
             self.assertEqual(stdout, test_string)
             self.assertEqual(stderr, '')
 
+    @run_for_all_plugins
+    def test_exec_with_stdin_filelike_bytes(self, custom_transport):
+        """Test command execution with a stdin from filelike of type bytes.
+
+        I test directly the exec_command_wait_bytes function; I also pass some non-unicode
+        bytes to check that there is no internal implicit encoding/decoding in the code.
```

Comment:
Maybe nitpicky, but maybe there is another way to express this? I feel like "checking that there is no internal implicit encoding/decoding" is not really possible (I understand one can only unit test for behavior, what the method does internally is up to them). This also goes for the other test.

I guess the behaviour here is that we want to check that bytes that are non unicode are still accepted, and maybe add a comment that this should be true for any encoding but we can't test for all of them (and also I guess some of these can accept any byte?).